src/modules/core/components/NetworkStatus/NetworkStatus.tsx

```diff
@@ -0,0 +1,185 @@
+import styles from './NetworkStatus.css';
+import {
+  useLatestRpcBlockQuery,
+  useColonyServerLivenessQuery,
+  useLatestSubgraphBlockQuery,
+  useReputationOracleLivenessQuery,
+} from '~data/index';
+
+import Popover from '~core/Popover';
+import Icon from '~core/Icon';
+import ExternalLink from '~core/ExternalLink';
+import { NETWORK_HEALTH } from '~externalUrls';
+import { MiniSpinnerLoader } from '~core/Preloaders';
+
+const MSG = defineMessages({
+  networkStatusButton: {
+    id: 'pages.NavigationWrapper.UserNavigation.networkStatusButton',
+    defaultMessage: `Network`,
+  },
+
+  networkHealthHeader: {
+    id: 'pages.NavigationWrapper.UserNavigation.networkHealthHeader',
+    defaultMessage: `Network health is {networkHealth}`,
+  },
+
+  networkHealthDescription: {
+    id: 'pages.NavigationWrapper.UserNavigation.networkHealthDescription',
+    defaultMessage: `{networkHealth, select,
+      poor
+        {You should be able to perform actions, however,
+        there is problem retrieving new information from the
+        chain. We are working to resolve this.}
+      critical
+        {You will have trouble performing actions and retrieving
+        information from the chain. We are working to resolve this.}
+        other {Network is healthy.}}`,
+  },
+});
+
+const NetworkStatus = () => {
+  enum NetworkHealthType {
+    Healthy = 'healthy',
+    Poor = 'poor',
+    Critical = 'critical',
+  }
+  const [networkHealth, setNetworkHealth] = useState<NetworkHealthType>(
+    NetworkHealthType.Healthy,
+  );
+
+  const networkCheckInterval = 120 * 1000; // 2 minutes
+
+  const {
+    data: latestRpcBlock,
+    error: latestRpcBlockError,
+  } = useLatestRpcBlockQuery({
+    pollInterval: networkCheckInterval,
+  });
+
+  const { data: isColonyServerAlive } = useColonyServerLivenessQuery({
+    pollInterval: networkCheckInterval,
+  });
+
+  const { data: isReputationOracleAlive } = useReputationOracleLivenessQuery({
+    pollInterval: networkCheckInterval,
+  });
+
+  const {
+    data: latestSubgraphBlock,
+    error: latestSubgraphBlockError,
+  } = useLatestSubgraphBlockQuery({
+    pollInterval: networkCheckInterval,
+  });
+
+  useEffect(() => {
+    const networkCheck = setInterval(async () => {
+      if (latestRpcBlockError) {
+        setNetworkHealth(NetworkHealthType.Critical);
+      } else if (
+        latestSubgraphBlockError ||
+        !isReputationOracleAlive?.isReputationOracleAlive ||
+        !isColonyServerAlive?.isServerAlive ||
+        (latestRpcBlock &&
+          latestSubgraphBlock &&
+          latestRpcBlock.latestRpcBlock >
+            latestSubgraphBlock.latestSubgraphBlock)
+      ) {
+        setNetworkHealth(NetworkHealthType.Poor);
+      } else {
+        // If everything is okay, set health to healthy (to correct for the previous state)
+        setNetworkHealth(NetworkHealthType.Healthy);
+      }
+    }, networkCheckInterval);
+    return () => clearInterval(networkCheck);
+  });
```

Comment:
Running these every second incurs a very serious performance penalty. Both in app memory, but especially in infrastrcture resources.

See demo: 
![demo-1m-requests-per-second](https://user-images.githubusercontent.com/1193222/163832261-6232eb6e-54f7-40dd-8a61-93ab68dd35d3.gif)

This is not necessarily a problem with your implementation as you just followed the specs, and more of a problem of specs not looking at the big picture.

For this we need to take a step back and come up with some sort of deffering or debouncing mechanism _(although debouncing is not really suitable here)_

We need to have a call with the Product team and come up with alternatives as I'm not comfortable pushing this into production.

Make sure to schedule a call w/ you, @arrenv and me to hash this out.